crates/core/src/table/mod.rs

```diff
@@ -1260,6 +1326,58 @@ mod tests {
         assert!(file_slice_2.log_files.is_empty());
     }
 
+    #[test]
```

Comment:
let's also add a couple of boundary conditions like n=0, end_timestamp=None.

Reply:
yup that can be handled in the extracted common helper function for splitting